packages/plugin-metrics/src/metrics.js

```diff
@@ -26,16 +26,75 @@ const Metrics = SparkPlugin.extend({
     return this.batcher.request(Object.assign({key}, value));
   },
 
-  submitClientMetrics(eventName, props) {
+  /**
+   * This corresponds to #sendSemiStructured() in the deprecated metrics handler
+   * @param eventName
+   * @param props
+   * @param preLoginId
+   */
+  submitClientMetrics(eventName, props, preLoginId) {
     const payload = {metricName: eventName};
     if (props.tags) {
       payload.tags = props.tags;
     }
     if (props.fields) {
       payload.fields = props.fields;
     }
-    return this.clientMetricsBatcher.request(payload);
+    if (props.type) {
+      payload.type = props.type;
+    }
+    if (preLoginId) {
+      const _payload = {
+        metrics: [
+          payload
+        ]
+      };
+      // Do not batch these because pre-login events occur during onboarding, so we will be partially blind
+      // to users' progress through the reg flow if we wait to persist pre-login metrics for people who drop off because
+      // their metrics will not post from a queue flush in time
+      this.postPreLoginMetric(_payload, preLoginId);
+    } else {
+      return this.clientMetricsBatcher.request(payload);
+    }
+  },
+
+
+  /**
+   * Issue request to alias a user's pre-login ID with their CI UUID
+   * @param {string} preLoginId
+   */
+  alias: function(preLoginId) {
+    return this.request({
+      method: `POST`,
+      api: `metrics`,
+      resource: `clientmetrics`,
+      headers: {
+        "X-Prelogin-UserId": preLoginId
+      },
+      body: {},
+      params: [
+        {
+          name: 'alias',
+          value: true
+        }
+      ]
+    });
+  },
+
+  postPreLoginMetric: function(payload, preLoginId) {
+    return this.request({
+      method: `POST`,
+      url: 'https://metrics-a.wbx2.com/metrics/api/v1/clientmetrics-prelogin',
+      headers: {
+        "X-Prelogin-UserId": preLoginId,
+        "content-type": "application/json",
```

Comment:
left side should be apostrophes, right side shoul dbe back ticks.

the content type is unnecessary because of `json:true` (set by default in http-core). 

@Cruig, is `x-prelogin-userid` in the cloudapps list of cors-allowed headers? 

Please make the header name lowercase.